src/Makefile

```diff
@@ -64,3 +64,13 @@
 # config.mk file.
 MAKEFILE_PATH = ../libavr32/asf/avr32/utils/make/Makefile.avr32.in
 include $(MAKEFILE_PATH)
+
+.PHONY: format
+format:
+	clang-format -style=file -i main.c
+	clang-format -style=file -i table.h
+	clang-format -style=file -i table.c
+	clang-format -style=file -i teletype.h
+	clang-format -style=file -i teletype.c
+	clang-format -style=file -i euclidean/euclidean.h
+	clang-format -style=file -i euclidean/euclidean.c
```

Comment:
Hadn't noticed that you're generating from Haskell. Neat!

Another consideration would be to add codegen to the build process and then not commit the generated C sources.  Of course, that requires all devs have haskell installed...  (Maybe that's a show-stopper.)

But anyway, cool!